sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+     * potentially deal with a higher precision value than what is supported by JavaScript.**
+     */
+    expiry: string;
+    /**
+     * the permissions for the acl policy
+     */
+    permission: string;
+}
+/**
+ * Properties of a blob
+ */
+export interface BlobProperties {
+    creationTime?: Date;
+    lastModified: Date;
+    etag: string;
+    /**
+     * Size in bytes
+     */
+    contentLength?: number;
+    contentType?: string;
+    contentEncoding?: string;
+    contentLanguage?: string;
+    contentMD5?: Uint8Array;
+    contentDisposition?: string;
+    cacheControl?: string;
+    blobSequenceNumber?: number;
+    /**
+     * Possible values include: 'BlockBlob', 'PageBlob', 'AppendBlob'
+     */
+    blobType?: BlobType;
+    /**
+     * Possible values include: 'locked', 'unlocked'
+     */
+    leaseStatus?: LeaseStatusType;
+    /**
+     * Possible values include: 'available', 'leased', 'expired', 'breaking', 'broken'
+     */
+    leaseState?: LeaseStateType;
+    /**
+     * Possible values include: 'infinite', 'fixed'
+     */
+    leaseDuration?: LeaseDurationType;
+    copyId?: string;
+    /**
+     * Possible values include: 'pending', 'success', 'aborted', 'failed'
+     */
+    copyStatus?: CopyStatusType;
+    copySource?: string;
+    copyProgress?: string;
+    copyCompletionTime?: Date;
+    copyStatusDescription?: string;
+    serverEncrypted?: boolean;
+    incrementalCopy?: boolean;
+    destinationSnapshot?: string;
+    deletedTime?: Date;
+    remainingRetentionDays?: number;
+    /**
+     * Possible values include: 'P4', 'P6', 'P10', 'P15', 'P20', 'P30', 'P40', 'P50', 'P60', 'P70',
+     * 'P80', 'Hot', 'Cool', 'Archive'
+     */
+    accessTier?: AccessTier;
+    accessTierInferred?: boolean;
+    /**
+     * Possible values include: 'rehydrate-pending-to-hot', 'rehydrate-pending-to-cool'
+     */
+    archiveStatus?: ArchiveStatus;
+    customerProvidedKeySha256?: string;
+    accessTierChangeTime?: Date;
+}
+/**
+ * An interface representing BlobMetadata.
+ */
+export interface BlobMetadata {
+    encrypted?: string;
+    /**
+     * Describes unknown properties. The value of an unknown property MUST be of type "string". Due
+     * to valid TS constraints we have modeled this as a union of `string | any`.
+     */
+    [property: string]: string | any;
+}
+/**
+ * An Azure Storage blob
+ */
+export interface BlobItem {
+    name: string;
+    deleted: boolean;
+    snapshot: string;
+    properties: BlobProperties;
+    metadata?: BlobMetadata;
+}
+/**
+ * An interface representing BlobFlatListSegment.
+ */
+export interface BlobFlatListSegment {
+    blobItems: BlobItem[];
+}
+/**
+ * An enumeration of blobs
+ */
+export interface ListBlobsFlatSegmentResponse {
+    serviceEndpoint: string;
+    containerName: string;
+    prefix?: string;
+    marker?: string;
+    maxResults?: number;
+    delimiter?: string;
```

Comment:
I dont' think delimiter comes back for a flat listing; only for a hierarchical listing.

Reply:
Should be fixed in swagger.